core/src/main/java/io/github/fourlastor/game/demo/turns/AttackMelee.java

```diff
@@ -0,0 +1,132 @@
+package io.github.fourlastor.game.demo.turns;
+
+import com.badlogic.gdx.math.Interpolation;
+import com.badlogic.gdx.math.MathUtils;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.math.Vector3;
+import com.badlogic.gdx.scenes.scene2d.actions.Actions;
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction;
+import com.badlogic.gdx.utils.Align;
+import dagger.assisted.Assisted;
+import dagger.assisted.AssistedFactory;
+import dagger.assisted.AssistedInject;
+import io.github.fourlastor.game.demo.state.GameState;
+import io.github.fourlastor.game.demo.state.unit.Unit;
+
+public class AttackMelee extends TurnState {
+
+    private final StateRouter router;
+
+    private final Unit source;
+    private final Unit target;
+    Vector2 originalPosition = new Vector2();
+    float moveDuration = 0.05f;
+    // Amount to scale the animation by.
+    Vector3 scale = new Vector3(1f / 16f, 1f, 1f);
+
+    // Base animation goes left-to-right.
+    Vector3[] positionsRelative = new Vector3[] {
+        new Vector3(-.25f, 0f, 0f),
+        new Vector3(-.125f, 0f, 0f),
+        new Vector3(0f, 0f, 0f),
+        new Vector3(2f, 0f, 0f),
+        new Vector3(8f, 0f, 0f),
+        new Vector3(8f, 0f, 0f),
+        new Vector3(0f, 0f, 0f),
+        new Vector3(0f, 0f, 0f),
+        new Vector3(0f, 0f, 0f),
+        new Vector3(0f, 0f, 0f),
+        new Vector3(-3f, 0f, 16f),
+        new Vector3(-3f, 0f, 8f),
+        new Vector3(-3f, 0f, 0f),
+        new Vector3(-3f, 0f, -8f),
+        new Vector3(-3f, 0f, -16f),
+    };
+
+    @AssistedInject
+    public AttackMelee(@Assisted Attack attack, StateRouter router) {
+        this.router = router;
+        this.source = attack.source;
+        this.target = attack.target;
+        this.originalPosition.set(source.getActorPosition());
+    }
+
+    /**
+     * TODO: this functionality can be used generically.
+     *
+     * Calculate the angle in degrees between two 2D vectors.
+     *
+     * @param source The source vector.
+     * @param target The target vector.
+     * @return The angle in degrees between the vectors.
+     */
+    public static float calculateAngle(Vector2 source, Vector2 target) {
+        float deltaX = target.x - source.x;
+        float deltaY = target.y - source.y;
+
+        float angleRad = MathUtils.atan2(deltaY, deltaX);
+        float angleDeg = MathUtils.radiansToDegrees * angleRad;
+
+        return angleDeg;
+    }
```

Comment:
I think this is equivalent to `source.angleDeg(target)`

Reply:
source.angleDeg(target) was not working for some reason.